### DISCUSSION TITLE
Styles for generated ToC

### DISCUSSION THREAD

**Discussion by warbe-maker** [allejo/jekyll-toc#50] [Q&A] [answered]
Supposed I'll get it running one day, is it possible to have a layout like the one offered by https://github.com/dafi/jekyll-toc-generator.
Which adds to my question: That's a js solution which you said is not possible ?!?

**Comment by allejo** [comment, marked as answer]
> is it possible to have a layout like the one offered by dafi/jekyll-toc-generator

Yes, it is possible. This snippet only focuses on generating the actual content for the table of contents. The design of it is left up entirely to the user of this project. All that's needed is some CSS and applying classes to elements :smile:

> Which adds to my question: That's a js solution which you said is not possible ?!?

I'm not sure where I said a JS solution isn't possible but that's not true. In the README I specify you can't use custom Jekyll plug-ins (written in Ruby) on GitHub Pages. I also specify that you _can_ use a JS approach, but if users have JS disabled or you're using automated link checkers that don't run JS, you'll run into problems.

See README:

> > GitHub Pages can't run custom Jekyll plug-ins so when generating Tables of Contents (TOCs), you're stuck with either a JavaScript solution or using kramdown's `{:toc}` option. However, by using `{:toc}`, you are forced to have that code next to your actual markdown and you can't place it in a layout. This means _every_. _single_. _post_. will need to have the snippet. If you choose the JavaScript approach, that's perfectly fine but what if JS is disabled on someone's browser or your page is just _really_ long and it becomes inefficient.
